backend/src/main/java/saci/domain/service/LevelService.java

```diff
@@ -0,0 +1,44 @@
+package saci.domain.service;
+
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import saci.domain.model.Level;
+import saci.domain.service.exceptions.AlreadyExistsException;
+import saci.domain.service.exceptions.CoefficientOverlapException;
+import saci.domain.service.validators.LevelValidator;
+import saci.infrastructure.LevelRepository;
+
+@Service
+@RequiredArgsConstructor
+public class LevelService {
+
+    private final LevelRepository levelRepository;
+
+    public Level createLevel(Level level) {
+        Optional<Level> optionalLevel = levelRepository.findByName(level.getName());
+        int overlappingLevels =
+                levelRepository.overlappingLevelsCounter(
+                        level.getRoleId(),
+                        Math.toIntExact(level.getMin_coefficient()),
+                        Math.toIntExact(level.getMax_coefficient()));
+        boolean validMinRange = LevelValidator.ValidMinRange(level);
+        boolean validMaxRange = LevelValidator.ValidMaxRange(level);
+        boolean overlappingLevelsValidation =
+                LevelValidator.OverlappingLevelsValidation(overlappingLevels);
+        if (validMaxRange && validMinRange && overlappingLevelsValidation) {
```

Comment:
You don't need to create a variable for each of the validations, call the methods directly in the if, just ident it carefully to not make it confusing, but great job here!